### ISSUE TITLE
Implement _exceptions list

### ISSUE THREAD

**Issue by pythons1980** [danielfm/pybreaker#20]
Hi All,

it would be nice if function `is_system_error` can work with `self._exceptions` as well. I am about to implement ES circuit breaker for just one exception (ES BROKEN CONNECTION) so that I have to create my own Breaker (subclassing CircuitBreaker) and rewrite function `is_system_error` instead of just define one exception in the new list `self._exceptions`.

Regards,
   Vojtech.

**Comment by sebastiandev** [comment]
Yes, I needed something similar and ended up using a callable that check that the error is of the type i want. But that functionality is only in the latest master, not in a release

**Comment by yairm210** [comment]
Now that some time has passed, it's simple to do this as sebastiandev has said

Including a test for this here, modeled after the existing tests in tests.py:

```python

    def test_allow_whitelist(self):
        """CircuitBreaker: it should allow the user to add a predicate function to determine excluded exceptions.
        """
        acceptableExceptions = [NotImplementedError]
        isUnacceptableException = lambda e: type(e) not in acceptableExceptions

        self.breaker = CircuitBreaker(exclude=[isUnacceptableException])


        def throwsAcceptableError(): raise NotImplementedError
        def throwsUnacceptableError(): raise LookupError

        self.assertRaises(NotImplementedError, self.breaker.call, throwsAcceptableError)
        self.assertEqual(1, self.breaker.fail_counter)

        self.assertRaises(LookupError, self.breaker.call, throwsUnacceptableError)
        self.assertEqual(0, self.breaker.fail_counter)

```

**Comment by AhsanSheraz** [comment]
This pattern is what Resilience4j calls `ignoreExceptions` on the Java side — exceptions that
propagate to the caller but are never recorded as circuit failures. It matters in practice:
client-side errors (validation, auth, quota) say nothing about the downstream service's health,
so counting them toward the failure threshold opens the circuit for the wrong reason.

In case it helps anyone landing here: [pyresilience](https://github.com/AhsanSheraz/pyresilience), which implements this as
`ignore_on` (since 0.4.0):

```python
from pyresilience import resilient, CircuitBreakerConfig

@resilient(circuit_breaker=CircuitBreakerConfig(
    failure_threshold=5,
    error_types=(Exception,),               # count everything...
    ignore_on=(AuthError, ValidationError), # ...except terminal client errors
))
def call_api(): ...
```

`ignore_on` takes precedence over `error_types`, and the same option exists on the retry config
so those exceptions aren't retried either. Hope that's useful until this lands in pybreaker.
